pkg/nfd-master/nfd-master.go

```diff
@@ -825,6 +846,80 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.No
 	return nil
 }
 
+func (m *nfdMaster) nfdAPIUpdateAllNodeFeatureGroups() error {
+	klog.V(1).InfoS("updating All NodeFeatureGroups, will process all nodeFeatureGroupRules")
+
+	nodeFeatureGroups, err := m.nfdController.featureGroupLister.List(k8sLabels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to get nodeFeatureGroupRules resources: %w", err)
+	}
+	for _, nodeFeatureGroup := range nodeFeatureGroups {
+		m.updaterPool.nfgQueue.Add(nodeFeatureGroup.Name)
+	}
+
+	return nil
+}
+
+func (m *nfdMaster) nfdAPIUpdateNodeFeatureGroup(nodeFeatureGroupName string) error {
+	klog.V(1).InfoS("processing of nodeFeatureGroup", "nodeFeatureGroup", nodeFeatureGroupName)
+	nodeGroupValidator := make(map[string]bool)
+	var err error
+
+	kubeconfig, err := utils.GetKubeconfig(m.args.Kubeconfig)
+	if err != nil {
+		return err
+	}
+	nfdClient := nfdclientset.NewForConfigOrDie(kubeconfig)
```

Comment:
We should get the cli as an arg to the function

Reply:
done